src/Mod/PartDesign/Gui/TaskExtrudeParameters.cpp

```diff
@@ -389,6 +389,82 @@ void TaskExtrudeParameters::addAxisToCombo(App::DocumentObject* linkObj, std::st
         lnk.setValue(linkObj, std::vector<std::string>(1, linkSubname));
 }
 
+void TaskExtrudeParameters::setCheckboxes(int index, Modes mode)
+{
+    // disable/hide everything unless we are sure we don't need it
+    // exception: the direction parameters are in any case visible
+    bool isLengthEditVisible = false;
+    bool isLengthEdit2Visible = false;
+    bool isOffsetEditVisible = false;
+    bool isOffsetEditEnabled = true;
+    bool isMidplaneEnabled = false;
+    bool isMidplaneVisible = false;
+    bool isReversedEnabled = false;
+    bool isFaceEditEnabled = false;
+
+    if (mode == Modes::Dimension) {
+        isLengthEditVisible = true;
+        ui->lengthEdit->selectNumber();
+        QMetaObject::invokeMethod(ui->lengthEdit, "setFocus", Qt::QueuedConnection);
+        isMidplaneVisible = true;
+        isMidplaneEnabled = true;
+        // Reverse only makes sense if Midplane is not true
+        isReversedEnabled = !ui->checkBoxMidplane->isChecked();
+    }
+    else if (mode == Modes::ThroughAll) {
+        isOffsetEditVisible = true;
+        isOffsetEditEnabled = false; // offset may have some meaning for through all but it doesn't work
+        isMidplaneEnabled = true;
+        isReversedEnabled = !ui->checkBoxMidplane->isChecked();
+    }
+    else if (mode == Modes::ToFirst) {
+        isOffsetEditVisible = true;
+        isReversedEnabled = true;
+    }
+    else if (mode == Modes::ToLast) {

```

Comment:
ToLast is equal to ThroughAll and thus this else block is dead code. So, what's still needed is the extrude type (Pad or Pocket) to make sure ThroughAll is only handled for Pockets and ToLast only for Pads.